provider/openstack/provider.go

```diff
@@ -1779,46 +1779,36 @@ func (e *Environ) Instances(ctx context.ProviderCallContext, ids []instance.Id)
 	if len(ids) == 0 {
 		return nil, nil
 	}
-	// Make a series of requests to cope with eventual consistency.
-	// Each request will attempt to add more instances to the requested
-	// set.
-	var foundServers []nova.ServerDetail
-	for a := shortAttempt.Start(); a.Next(); {
-		var err error
-		foundServers, err = e.listServers(ctx, ids)
-		if err != nil {
-			logger.Debugf("error listing servers: %v", err)
-			if !IsNotFoundError(err) {
-				handleCredentialError(err, ctx)
-				return nil, err
-			}
-		}
-		if len(foundServers) == len(ids) {
-			break
```

Comment:
It is a little surprising to be removing what is effectively giving some time  for things to come up.
Is it because we're handling a retry at a higher level and don't want to double retry? Do all calls to `Instances()` have that same behavior? It feels like a function that gets used in a few places.

Reply:
This same pattern is applied in the Azure and EC2 providers, but others don't do it. One caller of the method already has it's own retry strategy at a higher level.

The O7k logic will actually not return from instance creation until it verifies that the node is up and available, so this doesn't appear to be useful for eventual consistency, given that if an instance has an ID, is was verified as being up prior.

I think we should actually do an audit, remove the retry strategy from providers that use it, then implement it at a higher level if required. It certainly isn't useful for the instance-poller scenario.